text/0015-gas-payments.md

```diff
@@ -0,0 +1,90 @@
+# Casper Payments: Gas Payment Concepts
+
+## High level view
+This is the first of a few CEPs on Casper Payment model. 
+The motivation for this work is to make Casper much more user and dev 
+friendly than it is now. There are a few topics here, that mix. The change 
+to one of them might cause the domino effect on others, but we'll 
+try to chunk them into logical units.
+
+#### Token Standard 
```

Comment:
Maybe you could go into a bit of detail about the flexibilities and inflexibilities of current payment code, which, according to my prior understanding, gave us significantly more flexibility compared to ETH? Or was I wrong? I never really looked into it closely.

Reply:
A few months ago I have tested what's possible to achieve with payment codes. I tried to write a [contract](https://github.com/CasperLabs/contract-based-payment/blob/ECO-229-payment-code/contract/src/lib.rs), that pays for the execution. First problem was that it required user to have the minimal amount of tokens on in the main purse. The second thing was a problem with refunds. The system returns unused tokens to the caller, not to the purse that payed for the execution. Of course, it could be fixed and we could extend payment codes with more features, but that requires a good design that prevents spam attack, which we don't have at the moment.

My statement is that there is no way to write something different than the standard payment with a current implementation. I would love someone to prove me wrong.